includes/Core/Admin/Screens.php

```diff
@@ -359,16 +371,62 @@ private function get_screens() {
 			new Screen(
 				self::PREFIX . 'dashboard',
 				array(
-					'title'            => __( 'Dashboard', 'google-site-kit' ),
-					'capability'       => Permissions::VIEW_DASHBOARD,
-					'enqueue_callback' => function ( Assets $assets ) {
+					'title'               => __( 'Dashboard', 'google-site-kit' ),
+					'capability'          => Permissions::VIEW_DASHBOARD,
+					'enqueue_callback'    => function ( Assets $assets ) {
 						if ( $this->context->input()->filter( INPUT_GET, 'permaLink' ) ) {
 							$assets->enqueue_asset( 'googlesitekit-entity-dashboard' );
 						} else {
 							$assets->enqueue_asset( 'googlesitekit-main-dashboard' );
 						}
 					},
-					'render_callback'  => function ( Context $context ) {
+					'initialize_callback' => function ( Context $context ) {
+						$is_view_only = ! $this->authentication->is_authenticated();
+
+						if ( ! $is_view_only ) {
+							$initial_setup_settings      = ( new Initial_Setup_Settings( $this->user_options ) )->get();
+							$is_analytics_setup_complete = $initial_setup_settings['isAnalyticsSetupComplete'];
+
+							if ( false === $is_analytics_setup_complete ) {
+								$is_analytics_connected = $this->modules->is_module_connected( 'analytics-4' );
+
+								if ( $is_analytics_connected ) {
+									wp_safe_redirect(
+										$context->admin_url(
+											'user-input',
+											array(
+												'showProgress' => true,
+											)
+										)
+									);
+
+									exit;
+								} else {
+									$slug = $context->input()->filter( INPUT_GET, 'slug' );
+									$show_progress = (bool) $context->input()->filter( INPUT_GET, 'showProgress' );
+									$re_auth = (bool) $context->input()->filter( INPUT_GET, 'reAuth' );
```

Comment:
Same here. Use `FILTER_VALIDATE_BOOLEAN` when reading the values back.
```suggestion
									$re_auth = $context->input()->filter( INPUT_GET, 'reAuth', FILTER_VALIDATE_BOOLEAN );
```